### PR TITLE
Updated Docstrings and doctests

### PR DESCRIPTION
**Pull request by Dsantra92**
Continuation of  pull request : [1079](https://github.com/FluxML/Flux.jl/pull/1079) .The one hot vector and batch matrix docstrings and doc tests have been updated to change the boolean type to numeric type.@janEbert.

### PR Checklist

- [ ] Tests are added
- [ ] Entry in NEWS.md
- [x] Documentation, if applicable
- [ ] Final review from `@MikeInnes` or `@dhairyagandhi96` (for API changes).

### CODE REVIEW

**Comment by janEbert** [comment]
As I mentioned [here](https://github.com/FluxML/Flux.jl/pull/1079#issuecomment-617689702) in the latter half, you _could_ also change the code blocks in [docs/src/data/onehot.md](docs/src/data/onehot.md) so they get doctested to future-proof against mistakes. Be aware that you need to [handle](https://juliadocs.github.io/Documenter.jl/stable/man/doctests/#Preserving-Definitions-Between-Blocks-1) the `using` imports over successive blocks if you decide to implement these changes as well.

File: src/onehot.jl [hunk 1]
```diff
@@ -45,11 +45,16 @@ cudaconvert(x::OneHotMatrix{<:CuArray}) = OneHotMatrix(x.height, cudaconvert(x.d
 
 """
     onehot(l, labels[, unk])
```
Comment (CarloLucibello) [inline src/onehot.jl:47]: newline here

File: src/onehot.jl [hunk 2]
```diff
@@ -45,11 +45,16 @@ cudaconvert(x::OneHotMatrix{<:CuArray}) = OneHotMatrix(x.height, cudaconvert(x.d
...
+Create a `OneHotVector` based on `l` and `labels`
+
+If `l` is present in `k`th index of `labels` the function returns a `OneHotVector`
+with `1` in `k`th position and `0` in other in other indexes.
+
+If `l` is not found in labels and  `unk` is present, the function returns
+`onehot( unk, labels)`
+
+Otherwise the fucntion returns an error message.
```
Comment (CarloLucibello) [inline src/onehot.jl:56]:
```suggestion
Otherwise the function returns an error message.
```

File: src/onehot.jl [hunk 3]
```diff
@@ -45,11 +45,16 @@ cudaconvert(x::OneHotMatrix{<:CuArray}) = OneHotMatrix(x.height, cudaconvert(x.d
 
 """
     onehot(l, labels[, unk])
+Create a `OneHotVector` based on `l` and `labels`
+
+If `l` is present in `k`th index of `labels` the function returns a `OneHotVector`
+with `1` in `k`th position and `0` in other in other indexes.
+
+If `l` is not found in labels and  `unk` is present, the function returns
+`onehot( unk, labels)`
```
Comment (CarloLucibello) [inline src/onehot.jl:54]:
```suggestion
`onehot(unk, labels)`.
```

File: src/onehot.jl [hunk 4]
```diff
@@ -83,6 +88,9 @@ end
 
 Create a `OneHotMatrix` with a batch of labels based on the
 possible set of `labels`.
+If every element of `ls` is present in labels it return a matrix with columns as
+one-hot vectors.The `k`th column of matrix has the value of onehot(ls[k], labels).
```
Comment (CarloLucibello) [inline src/onehot.jl:92]:
```suggestion
If every element of `ls` is present in labels, it return a matrix with one-hot vectors as columns. The `k`th column of the matrix has value of `onehot(ls[k], labels)`.
```

**Comment by CarloLucibello** [comment]
something has gone wrong here in the rebase

**Comment by Dsantra92** [comment]
@CarloLucibello yes I felt that,I merged those suggestions and something went wrong..Later I noticed that someone just updated the master before I pulled this.I panicked and did some pulled some wrong set of reverts.I want to finish this thing so I was asking how could I resolve this mess.

**Comment by CarloLucibello** [comment]
What I usually do to rebase a PR is 
```julia
git rebase -i origin/master
```
You probably used merge instead here. I think the easiest solution is to close this PR and open a new one

**Comment by Dsantra92** [comment]
okay

**Pull request closed without merging**